### PR TITLE
Psr0Fixer, Psr4Fixer - ignore "new class" syntax

### PR DESCRIPTION
**Pull request by dg**
Adds support for PHP 7 anonymous classes.

### CODE REVIEW

File: src/Fixer/Basic/Psr4Fixer.php [hunk 1]
```diff
@@ -70,7 +70,12 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
 
                 $classyIndex = $tokens->getNextMeaningfulToken($index);
                 $classyName = $tokens[$classyIndex]->getContent();
-            }
+
+                $prevToken = $tokens[$tokens->getPrevMeaningfulToken($index)];
+                if ($prevToken->isGivenKind(T_NEW)) {
+                    return;
```
Comment (keradus) [inline src/Fixer/Basic/Psr4Fixer.php]: can't determining that class is anonymous be done before determining it;'s name ?

**Comment by keradus** [comment]
Hi, thanks for PR !
please, could you retarget it against 2.2 base branch ?

**Comment by dg** [comment]
Done

**Comment by keradus** [comment]
nope, base branch is still master

**Comment by keradus** [comment]
please fix issues reported by CIs

**Comment by dg** [comment]
Fixed :)

File: tests/Fixer/Basic/Psr4FixerTest.php [hunk 2]
```diff
@@ -162,6 +162,27 @@ class Psr4Fixer {}
         $this->doTest($expected, null, $file);
     }
 
+    public function testIgnoreAnonymousClass()
```
Comment (SpacePossum) [inline tests/Fixer/Basic/Psr4FixerTest.php]: please add the PHPDOc;
```php
    /**
     * @requires PHP 7.0
     */
```

as linting of the tests code fails for PHP5.6.

Comment (dg) [inline tests/Fixer/Basic/Psr4FixerTest.php]: done

File: tests/Fixer/Basic/Psr4FixerTest.php [hunk 3]
```diff
@@ -163,6 +163,30 @@ class Psr4Fixer {}
     }
 
     /**
+     * @requires PHP 7.0
+     */
+    public function testIgnoreAnonymousClass()
```
Comment (keradus) [inline tests/Fixer/Basic/Psr4FixerTest.php]: please add respective test for `Psr0Fixer`

**Comment by keradus** [comment]
I have updated your branch with 2.2 to have newest CircleCI config

**Comment by dg** [comment]
@keradus It is a big surprise for me that you can write to my branch. But now I see 3 commits there and I do not know if it's okay.

**Comment by SpacePossum** [comment]
@dg 
On the right side of the PR there is a checkbox to (not) allow maintainers to make changes to you work.
Just a little FYI for the future ;)

**Comment by dg** [comment]
I see it, great!

**Comment by keradus** [comment]
From a maintainer side, you even have a button "Update branch" (near "Merge PR") so I could merge base branch (here 2.2) to your branch directly from website.

>  But now I see 3 commits there and I do not know if it's okay.

that's perfectly fine, I will squash them during merging

**Comment by keradus** [comment]
please fix tests ;)

File: src/Fixer/Basic/Psr4Fixer.php [hunk 4]
```diff
@@ -68,6 +68,16 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
                     return;
                 }
 
+                $prevToken = $tokens[$tokens->getPrevMeaningfulToken($index)];
+                if ($prevToken->isGivenKind(T_NEW)) {
+                    return;
+                }
+
+                $prevToken = $tokens[$tokens->getPrevMeaningfulToken($index)];
```
Comment (SpacePossum) [inline src/Fixer/Basic/Psr4Fixer.php]: double code now ;)

**Comment by keradus** [comment]
Thank you @dg.

**Pull request closed without merging**